src/damavand/base/controllers/base_controller.py

```diff
@@ -60,6 +61,21 @@ def resource(self) -> PulumiResource:
 
         raise NotImplementedError()
 
+    @property
+    def environment(self) -> Environment:
+        """Return the that controller is being executed in."""
+
+        if env := self.extra_args.get("environment"):
```

Comment:
Nitpick: rename `extra_args` to `keyword_args` to follow industry standard.

Reply:
@bramelfrink good suggestion. There are a few things that needs to be changed with in the core API interface. I will take it into a separated issue with the other enhancements to the core API.